File: leetcode/Arrays_Hash/Sort Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        low = mid = 0
        end = n - 1
        while mid <= end:
            if nums[mid] == 0:
                nums[mid], nums[low] = nums[low], nums[mid]
                low += 1
                mid += 1
            elif (nums[mid] == 1):
                mid += 1
            else:
                nums[mid], nums[end] = nums[end], nums[mid]
                end -= 1

File: leetcode/Arrays_Hash/Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low = 0;
        int mid = 0;
        int end = nums.size() - 1;
        while (mid <= end){
            if (nums[mid] == 0){
                int temp = nums[low];
                nums[low] = nums[mid];
                nums[mid] = temp;
                mid ++;
                low ++;
            }
            else if (nums[mid] == 1){
                mid ++;
            }
            else {
                int temp = nums[end];
                nums[end] = nums[mid];
                nums[mid] = temp;
                end --;
            }
        }
    }
};